bodhi/server/tasks/handle_side_and_related_tags.py

```diff
@@ -20,67 +20,49 @@
 import logging
 import typing
 
-import koji  # noqa: 401
+from bodhi.server import buildsys
 
-from bodhi.server import buildsys, util
-from bodhi.server.models import Update
 
 log = logging.getLogger(__name__)
 
 
-def main(aliases: typing.List[str], from_tag: str):
+def main(builds: typing.List[str],
+         pending_signing_tag: str,
+         from_tag: str,
+         pending_testing_tag: typing.Optional[str]):
     """Handle side-tags and related tags for updates in Koji.
 
     Args:
-        aliases: one or more Update alias
-        from_tag: the tag into which the builds were built
+        builds: a list of builds to be tagged.
+        pending_signing_tag: the pending signing tag to apply on the builds.
+        from_tag: the tag into which the builds were built.
+        pending_testing_tag: the pending_testing_tag to create if not None.
     """
-    db_factory = util.transactional_session_maker()
-    koji = buildsys.get_session()
     try:
-        with db_factory() as db:
-            for alias in aliases:
-                update = db.query(Update).filter_by(alias=alias).first()
-                if update is not None:
-                    handle_tagging(update, from_tag, koji)
-    except Exception:
-        log.exception("There was an error handling side-tags updates")
-    finally:
-        db_factory._end_session()
+        koji = buildsys.get_session()
+        koji.multicall = True
 
+        if pending_testing_tag is not None:
+            # Validate that <koji_tag>-pending-signing and <koji-tag>-testing-signing exists
+            # if not create them.
+            if not koji.getTag(pending_signing_tag):
+                log.info(f"Create {pending_signing_tag} in koji")
+                koji.createTag(pending_signing_tag, parent=from_tag)
+            if not koji.getTag(pending_testing_tag):
+                log.info(f"Create {pending_testing_tag} in koji")
+                koji.createTag(pending_testing_tag, parent=from_tag)
+                koji.editTag2(pending_testing_tag, perm="autosign")
+        else:
+            # If we don't provide a pending_testing_tag, then we have merged the
+            # side tag into the release pending_signing tag.
+            # We can remove the side tag.
+            koji.removeSideTag(from_tag)
```

Comment:
I want to double check that we can call this before actually tagging the builds.

Reply:
So I have checked this and that will remove the tag from all the builds tagged with it. Since at this point the list of builds is persisted in bodhi. I believe this is ok to do that step here.

@pypingou do you any issue with that ?

Reply:
Since we have merged the side-tags into the pending_signing the builds shouldn't end up tag-less, so :+1: this should be fine